nemo/collections/tts/data/datalayers.py

```diff
@@ -0,0 +1,122 @@
+# limitations under the License.
+from dataclasses import dataclass
+from typing import Dict, Optional, Union
+
+import torch
+from omegaconf import MISSING, DictConfig, OmegaConf
+
+from nemo.collections.asr.parts import collections, parsers
+from nemo.collections.asr.parts.segment import AudioSegment
+
+
+@dataclass
+class AudioDatasetConfig:
+    manifest_filepath: str = MISSING
+    n_segments: int = MISSING
+    max_duration: Optional[float] = None
+    min_duration: Optional[float] = None
+    trim: Optional[bool] = False
+
+
+class AudioDataset(torch.utils.data.Dataset):
+    def __init__(
+        self,
+        manifest_filepath: Union[str, 'pathlib.Path'],
+        n_segments: int,
+        max_duration: Optional[float] = None,
+        min_duration: Optional[float] = None,
+        trim: Optional[bool] = False,
+    ):
+        """
+        Mostly compliant with nemo.collections.asr.data.datalayers.AudioToTextDataset except it only returns Audio
+        without text. Dataset that loads tensors via a json file containing paths to audio files, transcripts, and
+        durations (in seconds). Each new line is a different sample. Note that text is required, but is ignored for
+        AudioDataset. Example below:
+        {"audio_filepath": "/path/to/audio.wav", "text_filepath":
+        "/path/to/audio.txt", "duration": 23.147}
+        ...
+        {"audio_filepath": "/path/to/audio.wav", "text": "the
+        transcription", "offset": 301.75, "duration": 0.82, "utt":
+        "utterance_id", "ctm_utt": "en_4156", "side": "A"}
+        Args:
+            manifest_filepath (str, Path): Path to manifest json as described above. Can be comma-separated paths
+                such as "train_1.json,train_2.json" which is treated as two separate json files.
+            n_segments (int): The length of audio in samples to load. For example, given a sample rate of 16kHz, and
+                n_segments=16000, a random 1 second section of audio from the clip will be loaded. The section will
+                be randomly sampled everytime the audio is batched. Can be set to -1 to load the entire audio.
+            max_duration (float): If audio exceeds this length in seconds, it is filtered from the dataset.
+                Defaults to None, which does not filter any audio.
+            min_duration(float): If audio is less than this length in seconds, it is filtered from the dataset.
+                Defaults to None, which does not filter any audio.
+            trim (bool): Whether to use librosa.effects.trim on the audio clip
+        """
+
+        self.collection = collections.ASRAudioText(
+            manifests_files=manifest_filepath.split(','),
+            parser=parsers.make_parser(),
+            min_duration=min_duration,
+            max_duration=max_duration,
+        )
+        self.trim = trim
+        self.n_segments = n_segments
+
+    def collate_fn(self, batch):
+        return self._collate_fn(batch)

```

Comment:
What base class? This inherits from `torch.utils.data.Dataset`.
My rationale for this was so that I stop accessing _collate_fn when defining the dataloader

Reply:
Oh you can inherit Nemo Dataset class, its exactly the torch Dataset class with typecheck conveniences, but its alright not to use it since this is in experimental stage. 

The only benefit of inheriting the Nemo Dataset class is that if you override the `_collate_fn(self, batch)` method, and define `output_types()` then the collated result will get type checked. Not really necessary during research phase.